registrations/views.py

```diff
@@ -332,39 +334,15 @@ def post(self, request):
         serializer.is_valid(raise_exception=True)
 
         post_data = self.build_jembi_helpdesk_json(serializer.validated_data)
-        try:
-
-            source = Source.objects.get(user=self.request.user.id)
-
-            endpoint = "helpdesk"
-            if source.name == "NURSE Helpdesk App":
-                endpoint = "nc/helpdesk"
-                post_data["type"] = 12  # NC Helpdesk
-
-            result = requests.post(
-                urljoin(settings.JEMBI_BASE_URL, endpoint),
-                headers={"Content-Type": "application/json"},
-                data=json.dumps(post_data),
-                auth=(settings.JEMBI_USERNAME, settings.JEMBI_PASSWORD),
-                verify=False,
-            )
-            result.raise_for_status()
-        except (requests.exceptions.HTTPError,) as e:
-            if e.response.status_code == 400:
-                logger.warning(
-                    "400 Error when posting to Jembi.\n"
-                    "Response: %s\nPayload:%s"
-                    % (e.response.text, json.dumps(post_data))
-                )
-                return Response(
-                    "Error when posting to Jembi. Body: %s Payload: %r"
-                    % (e.response.content, post_data),
-                    status=status.HTTP_400_BAD_REQUEST,
-                )
-            else:
-                raise e
+        source = Source.objects.get(user=self.request.user.id)
+        endpoint = "helpdesk"
+        if source.name == "NURSE Helpdesk App":
+            endpoint = "nc/helpdesk"
+            post_data["type"] = 12  # NC Helpdesk
+        jembi_url = urljoin(settings.JEMBI_BASE_URL, endpoint)
+        raise_status = request_to_jembi_api(jembi_url, post_data)
```

Comment:
We probably don't need to return the `raise_for_status` here. Any error would cause an exception triggering our new retry strategy.

On success we can just return the 200 as it was, because the only way it will get to the return line is if nothing went wrong.